### PR TITLE
Support Filament 2.x

### PR DESCRIPTION
**Pull request by iantasker**
- Refactored to work with Filament admin v2.x

### CODE REVIEW

**Comment by jszobody** [comment]
Thanks for this! I left a couple comments.

File: src/CanImpersonateUsers.php [hunk 1]
```diff
@@ -12,16 +12,16 @@ trait CanImpersonateUsers
 {
     public function impersonate($key)
     {
-        $target = $this->getQuery()->find($key);
+        $target = $this->getTableQuery()->find($key);
         $this->authorizeImpersonate($target);
 
         app(ImpersonateManager::class)->take(
-            auth()->user(), $target, isset($this->guard) ? $this->guard : config('filament-impersonate.default_guard')
+            auth()->user(), $target, config('filament-impersonate.default_guard')
```
[the remark this answers is not in the document]
Comment (iantasker) [reply src/CanImpersonateUsers.php]: I have renamed to `guard`.

**Comment by iantasker** [comment]
@jszobody Would it be possible to get this merged?

**Comment by jszobody** [comment]
@iantasker Have you already installed this branch in a filament 2 app and verified it all works properly? I haven't had time to do anything with filament 2 yet (hope to soon!). If you're confident it's set to go I'll tag a 2.0 release.

**Pull request merged**